images/linux/ubuntu2004.json

```diff
@@ -0,0 +1,324 @@
+{
+    "variables": {
+        "commit_url": "{{env `COMMIT_URL`}}",
+        "client_id": "{{env `ARM_CLIENT_ID`}}",
+        "client_secret": "{{env `ARM_CLIENT_SECRET`}}",
+        "subscription_id": "{{env `ARM_SUBSCRIPTION_ID`}}",
+        "tenant_id": "{{env `ARM_TENANT_ID`}}",
+        "resource_group": "{{env `ARM_RESOURCE_GROUP`}}",
+        "storage_account": "{{env `ARM_STORAGE_ACCOUNT`}}",
+        "temp_resource_group_name": "{{env `TEMP_RESOURCE_GROUP_NAME`}}",
+        "location": "{{env `ARM_RESOURCE_LOCATION`}}",
+        "virtual_network_name": "{{env `VNET_NAME`}}",
+        "virtual_network_resource_group_name": "{{env `VNET_RESOURCE_GROUP`}}",
+        "virtual_network_subnet_name": "{{env `VNET_SUBNET`}}",
+        "private_virtual_network_with_public_ip": "{{env `PRIVATE_VIRTUAL_NETWORK_WITH_PUBLIC_IP`}}",
+        "image_folder": "/imagegeneration",
+        "commit_file": "/imagegeneration/commit.txt",
+        "imagedata_file": "/imagegeneration/imagedata.json",
+        "metadata_file": "/imagegeneration/metadatafile",
+        "installer_script_folder": "/imagegeneration/installers",
+        "helper_script_folder": "/imagegeneration/helpers",
+        "vm_size": "Standard_DS2_v2",
+        "capture_name_prefix": "packer",
+        "image_version": "dev",
+        "image_os": "ubuntu20",
+        "github_feed_token": null,
+        "go_default": "1.14",
+        "go_versions": "1.14"
+    },
+    "sensitive-variables": ["client_secret", "github_feed_token"],
+    "builders": [
+        {
+            "type": "azure-arm",
+            "client_id": "{{user `client_id`}}",
+            "client_secret": "{{user `client_secret`}}",
+            "subscription_id": "{{user `subscription_id`}}",
+            "tenant_id": "{{user `tenant_id`}}",
+
+            "location": "{{user `location`}}",
+            "vm_size": "{{user `vm_size`}}",
+            "resource_group_name": "{{user `resource_group`}}",
+            "storage_account": "{{user `storage_account`}}",
+            "temp_resource_group_name": "{{user `temp_resource_group_name`}}",
+            "capture_container_name": "images",
+            "capture_name_prefix": "{{user `capture_name_prefix`}}",
+            "virtual_network_name": "{{user `virtual_network_name`}}",
+            "virtual_network_resource_group_name": "{{user `virtual_network_resource_group_name`}}",
+            "virtual_network_subnet_name": "{{user `virtual_network_subnet_name`}}",
+            "private_virtual_network_with_public_ip": "{{user `private_virtual_network_with_public_ip`}}",
+            "os_type": "Linux",
+            "image_publisher": "canonical",
+            "image_offer": "0001-com-ubuntu-server-focal",
```

Comment:
this is now `UbuntuServer` (they finally moved it) and sku is `18.04-LTS`. Full spec from Azure:

```
                    "imageReference": {
                        "publisher": "Canonical",
                        "offer": "UbuntuServer",
                        "sku": "18.04-LTS",
                        "version": "latest"
```

Reply:
It does not work for some reason
 InvalidParameter : The following list of images referenced from the deployment template are not found: Publisher: canonical, Offer: UbuntuServer, Sku: 20.04-LTS

Reply:
18.04 is already set to the 
```
            "image_publisher": "Canonical",
            "image_offer": "UbuntuServer",
            "image_sku": "18.04-LTS",
```

Reply:
@alepauly, Currently I can't find the spec for `Ubuntu 20.04`. The latest available version under that spec is 19_04/19_10.

```
PS > Get-AzVMImage -Location westus -PublisherName Canonical -Offer UbuntuServer -Sku "20.04-LTS"
Get-AzVMImage : Artifact: VMImage was not found.
ErrorCode: NotFound
ErrorMessage: Artifact: VMImage was not found.
```

Reply:
> It does not work for some reason
> ` InvalidParameter : The following list of images referenced from the deployment template are not found: Publisher: canonical, Offer: UbuntuServer, Sku: 20.04-LTS`
> ![image](https://user-images.githubusercontent.com/48208649/82553949-ac23f680-9b6d-11ea-9d33-40c6ef47224b.png)
> 

Must be my mistake, I had both 18.04 and 20.04 specs open and must have rushed. Sorry for the confusion